nucliadb/nucliadb/ingest/orm/brain.py

```diff
@@ -260,84 +268,111 @@ def delete_metadata(self, field_key: str, metadata: FieldComputedMetadata):
-            sentence = paragraph.sentences[sent_key]
-
-            sentence.ClearField("vector")  # clear first to prevent duplicates
-            sentence.vector.extend(vector.vector)
-
-            # we only care about start/stop position of the paragraph for a given sentence here
-            # the key has the sentence position
-            sentence.metadata.position.start = vector.start_paragraph
-            sentence.metadata.position.end = vector.end_paragraph
-
-            # does it make sense to copy forward paragraph values here?
-            sentence.metadata.position.page_number = (
-                paragraph.metadata.position.page_number
+            paragraph_key = FIELD_PARAGRAPH_ID.format(
+                rid=self.rid,
+                field_id=field_id,
+                paragraph_start=vector.start_paragraph,
+                paragraph_end=vector.end_paragraph,
             )
-            sentence.metadata.position.in_page = paragraph.metadata.position.in_page
-
-            sentence.metadata.page_with_visual = paragraph.metadata.page_with_visual
-
-            sentence.metadata.representation.file = (
-                paragraph.metadata.representation.file
+            sentence_key = FIELD_VECTOR_ID.format(
+                rid=self.rid,
+                field_id=field_id,
+                index=index,
+                vector_start=vector.start,
+                vector_end=vector.end,
             )
-            sentence.metadata.representation.is_a_table = (
-                paragraph.metadata.representation.is_a_table
+            self._apply_field_vector(
+                field_id,
+                paragraph_key,
+                sentence_key,
+                vector,
+                vector_dimension=vector_dimension,
             )
 
-            sentence.metadata.position.index = paragraph.metadata.position.index
-
         for split in replace_splits:
-            self.brain.sentences_to_delete.append(f"{self.rid}/{field_key}/{split}")
+            self.brain.sentences_to_delete.append(f"{self.rid}/{field_id}/{split}")
 
         if replace_field:
-            self.brain.sentences_to_delete.append(f"{self.rid}/{field_key}")
+            self.brain.sentences_to_delete.append(f"{self.rid}/{field_id}")
+
+    def _apply_field_vector(
+        self,
+        field_id: str,
+        paragraph_key: str,
+        sentence_key: str,
+        vector: utils_pb2.Vector,
+        *,
+        vector_dimension: Optional[int] = None,
+    ):
+        paragraph_pb = self.brain.paragraphs[field_id].paragraphs[paragraph_key]
+        sentence_pb = paragraph_pb.sentences[sentence_key]
+
+        sentence_pb.ClearField("vector")  # clear first to prevent duplicates
+
+        # cut vectors if a specific dimension is specified
+        if vector_dimension is not None:
+            sentence_pb.vector.extend(vector.vector[:vector_dimension])
+        else:
+            sentence_pb.vector.extend(vector.vector)
```

Comment:
nit: you may not be choosing the implementation above for clarity, but turns out that
```
a = [1,2,3]
a[:None] == [1,2,3]
```

Reply:
I think I'll stick with the conditional for clarity